application_sandbox/sample_application_framework/sample_application.h

```diff
@@ -415,6 +415,9 @@ class Sample {
     return base->depth_stencil_->get_raw_image();
   }
 
+  FrameData* GetFrameData(uint32_t frame_idx) {
```

Comment:
This makes me a bit nervous. It somewhat breaks the abstraction/safety net provided by the fact that you cannot reach outside your own frame data.

Reply:
I would instead, make your cross-frame data a part of the application data, and manage the number of buffers, etc yourself.